gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -1785,9 +1785,24 @@ gst_tensor_converter_parse_caps (GstTensorConverter * self,
       break;
     default:
     {
+      GstCaps *peer_caps = gst_pad_peer_query_caps (self->srcpad, NULL);
+      gboolean is_fixed = FALSE;
+
+      if (peer_caps) {
+        is_fixed = gst_caps_is_fixed (peer_caps);
+
+        if (is_fixed)
+          gst_tensors_config_from_structure (&config,
+              gst_caps_get_structure (peer_caps, 0));
+
+        gst_caps_unref (peer_caps);
+      }
+
       if (self->mode == _CONVERTER_MODE_CUSTOM_CODE) {
-        gst_tensors_config_init (&config);
+        if (is_fixed)
+          break;
```

Comment:
Q. Possibly moving this to line 1800?

Reply:
To register an external converter, moving to line 1800 is impossible in this code status.